devtools/x/src/build.rs

```diff
@@ -0,0 +1,155 @@
+    /// Package to build (see `cargo help pkgid`)
+    package: Vec<String>,
+    #[structopt(long)]
+    /// Alias for --workspace (deprecated)
+    all: bool,
+    #[structopt(long)]
+    /// Build all packages in the workspace
+    workspace: bool,
+    #[structopt(long, number_of_values = 1)]
+    /// Package to build (see `cargo help pkgid`)
+    exclude: Vec<String>,
+    #[structopt(long, short)]
+    /// Number of parallel jobs, defaults to # of CPUs
+    jobs: Option<u16>,
+    #[structopt(long)]
+    /// Build only this package's library
+    lib: bool,
+    #[structopt(long, number_of_values = 1)]
+    /// Build only the specified binary
+    bin: Vec<String>,
+    #[structopt(long)]
+    /// Build all binaries
+    bins: bool,
+    #[structopt(long, number_of_values = 1)]
+    /// Build only the specified example
+    example: Vec<String>,
+    #[structopt(long)]
+    /// Build all examples
+    examples: bool,
+    #[structopt(long, number_of_values = 1)]
+    /// Build only the specified test target
+    test: Vec<String>,
+    #[structopt(long)]
+    /// Build all tests
+    tests: bool,
+    #[structopt(long, number_of_values = 1)]
+    /// Build only the specified bench target
+    bench: Vec<String>,
+    #[structopt(long)]
+    ///  Build all benches
+    benchs: bool,
+    #[structopt(long = "all-targets")]
+    /// Build all targets
+    all_targets: bool,
+    #[structopt(long)]
+    /// Build artifacts in release mode, with optimizations
+    release: bool,
+    #[structopt(long)]
+    /// Build artifacts with the specified profile
+    profile: Option<String>,
+    #[structopt(long, number_of_values = 1)]
+    /// Space-separated list of features to activate
+    features: Vec<String>,
+    #[structopt(long = "all-features")]
+    /// Activate all available features
+    all_features: bool,
+    #[structopt(long = "no-default-features")]
+    /// Do not activate the `default` feature
+    no_default_features: bool,
+    #[structopt(long)]
+    /// TRIPLE
+    target: Option<String>,
+    #[structopt(long = "target-dir", parse(from_os_str))]
+    /// Directory for all generated artifacts
+    target_dir: Option<OsString>,
+    #[structopt(long = "out-dir", parse(from_os_str))]
+    /// Copy final artifacts to this directory (unstable)
+    out_dir: Option<OsString>,
+    #[structopt(long = "manifest-path", parse(from_os_str))]
+    /// Path to Cargo.toml
+    manifest_path: Option<OsString>,
+    #[structopt(long = "message-format")]
+    /// Error format
+    message_format: Option<String>,
+    #[structopt(long = "build-plan")]
+    /// Output the build plan in JSON (unstable)
+    build_plan: bool,
+    //TODO: support -vv?
+    #[structopt(long, short)]
+    ///  Use verbose output (-vv very verbose/build.rs output)
+    verbose: bool,
+    #[structopt(long, possible_values = &Coloring::variants(), default_value="Auto")]
+    ///  Coloring: auto, always, never
+    coloring: Coloring,
```

Comment:
We probably don't need to support 100% of these things from the start, but this looks like a complete list